enhancements/security/openshift-image-policy.md

```diff
@@ -0,0 +1,359 @@
+* Users may add additional policies if they want, to add additional requirements as they see fit, including for the `quay.io/openshift-release-dev` scope.
+
+## Motivation
+
+As an OpenShift cluster administrator, I want to trust that the release image I'm running comes from Red Hat using the emerging industry standard for doing so: [Sigstore][].
+
+### User Stories
+
+* As a cluster administrator, I want to trust that the OpenShift release image I'm running comes from Red Hat, checking whenever a pod is launched from such an image, and not just at update-time.
+* As a cluster administrator, I want to be able to declare my own additional image policies, including restricting the use of the `quay.io/openshift-release-dev` scope.
+* As a Red Hat CI job maintainer or quality engineer, I have a way to disable Sigstore verification while testing releases before they have been signed.
+
+### Goals
+
+* Define an initial OpenShift ClusterImagePolicy that will be installed in all 4.17 clusters to restrict `quay.io/openshift-release-dev/ocp-release` to Red-Hat-signed images.
+
+### Non-Goals
+
+There are a number of possible next-steps that I'm punting on for now, as I try to nibble off the smallest possible slice of progress.
+This enhancement will not:
+
+* Figure out how to configure a ClusterImagePolicy for verifying a mirrored release image, although this will need to happen before we can use Sigstore signatures in disconnected/restricted-network environments.
+* Suggest changes to mirroring tools, although they will need to learn to mirror Sigstore signatures for use in disconnected/restricted-network environments.
+* Suggest changes to the ImageStream controller, although it will need to learn to mirror Sigstore signatures for verification when running pods from ImageStream sources, e.g. [must-gather][].
+* Suggest policies for `quay.io/openshift-release-dev/ocp-v4.0-art-dev`, although those would help tighten down security for other `openshift-*` namespaces.
+* Support testing unsigned releases as HyperShift hosted clusters on a management cluster that requires signed releases.
+
+## Proposal
+
+This enhancement proposes a new `openshift` ClusterImagePolicy that requires Red Hat signatures for `quay.io/openshift-release-dev/ocp-release` images.
+The new ClusterImagePolicy will be managed by the ClusterVersion operator and applied to all clusters by default (initially tech-preview, gated by the `ImagePolicy` feature gate ([currently `TechPreviewNoUpgrade`][feature-gate]).
+
+Internal test clusters, which need to run releases that are not signed by official Red Hat keys, will use a new install-config property `internalTestingImagePolicy: Disable` (also gated by the `ImagePolicy` feature gate, [precedent for a feature-gated install-config property][install-config-feature-gate-precedent]).
```

Comment:
:+1: to testing internally as well as having an opt out mechanism in the installer.

Reply:
What kind of internal testing are you thinking about?  I'm not sure what kept [ci-tools][1] from implementing the OpenPGP/GPG signatures before, but I'd guess it was:

* If we build this, we have to find some kind-of-secure way to handle signing of things built by not-particularly-hardened build clusters, and
* If we do get these signatures, we have [no GA way to point the cluster-version operator at them][2].

Sigstore hosted in the build cluster registries address the second, but the first would still need to get worked out.  I'd expect that to be mostly a DPTP ticket, without bumping into a lot of other teams, but I'm not clear enough on their setup, or on what's needed to host a Sigstore-signing-system, to know how much of a lift it would be or what it would entail.

[1]: https://github.com/openshift/ci-tools
[2]: https://issues.redhat.com/browse/OTA-1118

Reply:
turning this around a bit, is there a tracker for [these ClusterImagePolicy e2e tests][1]?  I'm not finding anything with [an origin search][2].

[1]: https://github.com/openshift/enhancements/blob/4668a0825c59739dfafd2ae661c16cf30f540946/enhancements/api-review/add-ClusterImagePolicy-and-ImagePolicy-for-Signature-Verification.md#test-plan
[2]: https://github.com/search?q=repo%3Aopenshift%2Forigin%20clusterimagepolicy&type=code

Reply:
@QiWang19 PTAL above

Reply:
https://issues.redhat.com/browse/OCPNODE-2183 This is a Jira tracker for ClusterImagePolicy e2e tests.